src/_guides/language/effective-dart/usage.md

```diff
@@ -245,87 +245,65 @@ void error([String? message = null]) {
-}
-
-// If you want null to be true:
-if (optionalThing?.isEnabled ?? true) {
-  print('Have enabled thing or nothing.');
-}
+if (nonNullableBool) { ... }
 {% endprettify %}
 
 {:.bad}
-<?code-excerpt "usage_bad.dart (convert-null-equals)"?>
 {% prettify dart tag=pre+code %}
-// If you want null to be false:
-if (optionalThing?.isEnabled == true) {
-  print('Have enabled thing.');
-}
-
-// If you want null to be true:
-if (optionalThing?.isEnabled != false) {
-  print('Have enabled thing or nothing.');
-}
+if (nonNullableBool == true) { ... }
 {% endprettify %}
 
-Both operations produce the same result and do the right thing, but `??` is
-preferred for three main reasons:
-
-*   The `??` operator signals that the code has something to do with `null`.
-
-*   The `== true` looks like a common mistake where the equality operator is
-    redundant and can be removed. That's true when the boolean expression on the
-    left will not produce `null`, but not when it can.
-
-*   The `?? false` and `?? true` clearly show what value will be used when the
-    expression is `null`. With `== true`, you have to think through the boolean
-    logic to realize that means that a `null` gets converted to *false*.
-
-**Exception:** Using a null-aware operator on a variable inside a condition
-doesn't promote the variable to a non-nullable type. If you want the variable
-to be promoted inside the body of the `if` statement, it might be better to use
-an explicit `!= null` check instead of `?.` followed by `??`:
+To evaluate a Boolean expression that *is nullable*, you should only use `??`
+or an explicit `!= null` check.
 
 {:.good}
-<?code-excerpt "usage_good.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message != null && message.isNotEmpty) {
-    // message is promoted to String.
-    return message.length;
-  }
+// If you want null to result in false:
+if (nullableBool ?? false) { ... }
 
-  return 0;
-}
+// If you want null to result in false
+// and you want the variable to type promote:
+if (nullableBool != null && nullableBool) { ... }
 {% endprettify %}
 
 {:.bad}
-<?code-excerpt "usage_bad.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message?.isNotEmpty ?? false) {
-    // message is not promoted to String.
-    return message!.length;
-  }
+// Error if null 
+if (nullableBool) { ... }
 
-  return 0;
-}
+// If you want null to be false:
+if (nullableBool == true) { ... }
 {% endprettify %}
 
+`if (nullableBool == true)` is a viable expression, but shouldn't be used for several reasons:
+
+* It doesn't indicate the code has anything to do with `null`.
+
+* Because it's not evidently `null` related, it can easily be mistaken for the non-nullable case,
+where the equality operator is redundant and can be removed.
+That’s only true when the boolean expression on the left has no chance of producing null, but not when it can.
```

Comment:
**suggestion:** This can go as is, but it could be tighter.

Consider:
* Removing the adverbs. You could say `...it's not clear that it relates to `null` (values? expressions?)...` and `...it can be mistaken...`
* Cutting the sentence into two: `You can remove the equality operator as it is redundant.`
* Simplify the language in the second sentence: `It's true when the left-side boolean expression can't produce a null value. It's false when it can't.`

Reply:
Skipping because:
> You could say ...it's not clear that it relates to null (values? expressions?)... and ...it can be mistaken...

"`null` related" is sufficient, really just related to null in any way is what's important, nothing specific about values/expressions/etc. 

"_easily_ mistaken", not just mistaken, is the issue here and why we're going out of the way to advise against using it even though it technically works. 

> Cutting the sentence into two: You can remove the equality operator as it is redundant.

The sentence is saying "it can be mistaken for the non-nullable case, _i.e. the case where the equality operator is redundant_". It's defining the clause before it, so cutting the sentence like that wouldn't make sense. 

> It's true when the left-side boolean expression can't produce a null value. It's false when it can't.

This reads much more awkward to me. Starting with "That's" indicates I'm talking about the condition I just described, while "It's" makes it sound like we're talking about the expression in question since it's referred to as "it" in every sentence prior. "It's false when it can't" is super confusing.